src/layers.c

```diff
@@ -184,7 +252,8 @@ handle_surface_commit(struct wl_listener *listener, void *data)
 	}
 	/* Process keyboard-interactivity change */
 	if (committed & WLR_LAYER_SURFACE_V1_STATE_KEYBOARD_INTERACTIVITY) {
-		process_keyboard_interactivity(layer);
+		struct seat *seat = &layer->server->seat;
+		layer_try_set_focus(seat, layer_surface);
 	}
```

Comment:
Just trying to think this through, if a layershell client was `INTERACTIVITY_NONE` before and then sets it to `ON_DEMAND`, we would steal focus from an active window or other layershell surface. That feels wrong.
It might be a different story for the initial mapping but this could be triggered anytime during the lifetime of the surface.

Reply:
Yes, I see what you mean. Hadn't thought of that.

So, `ON_DEMAND` should not do anything at the time of commit, only when receiving a cursor-button-press.

Let me patch and test with my local clients.

Reply:
Have patched && tested with `sfwbar`, `bemenu`, `tint` and `idesk-morfa`. I think this is better.

Reply:
Looks good. Sorry but I want to get all the details correct :p

What about the surface was `EXCLUSIVE`, had focus and then changed to `ON_DEMAND`?
I think that we have to let that one through as well, I am not sure if we can do it that easily though.. e.g. where to get the information that it was previously a exclusive surface.. Maybe we could just check for `ON_DEMAND` and compare it to the focused surface and if they match then call `layer_try_set_focus()`? Didn't think this through completely though.

Reply:
Good point - again - and as usual :smile:
Have tweaked patch.

Reply:
Can't spot anything else. Nice :)